File: 1433_Check_If_a_String_Can_Break_Another_String/solution.cpp
/*
1433. Check If a String Can Break Another String

Level: Medium

https://leetcode.com/problems/check-if-a-string-can-break-another-string
*/
#include <iostream>
#include <vector>
#include <string>
#include <stack>
#include <queue>
#include <unordered_map>
#include <unordered_set>
#include <numeric>
#include <algorithm>

#include "common/ListNode.h"
#include "common/TreeNode.h"

using namespace std;

/*
  Solution: 
*/

class Solution
{
public:
  bool checkIfCanBreak(string s1, string s2)
  {
    int count1 = 0, count2 = 0;
    if (s1.length() != s2.length())
      return false;

    int l = s1.length();
    sort(s1.begin(), s1.end());
    sort(s2.begin(), s2.end());
    
    for (int i = 0; i < l; i++)
    {
      if (s1[i] >= s2[i])
        count1++;
      if (s2[i] >= s1[i])
        count2++;
    }
    if (count1 == l || count2 == l)
      return true;
    return false;
  }
};

File: 1433_Check_If_a_String_Can_Break_Another_String/solution.py
'''
1433. Check If a String Can Break Another String

Level: Medium

https://leetcode.com/problems/check-if-a-string-can-break-another-string
'''
'''
  Solution: 
'''
class Solution:
    def checkIfCanBreak(self, s1: str, s2: str) -> bool:
      l1 = sorted(list(s1))
      l2 = sorted(list(s2))

      res = True
      for i,j in zip(l1,l2):
          if i < j:
              res = False
              break
      if res : return res

      res = True
      for i,j in zip(l1,l2):
          if i > j:
              res = False
              break

      return res